jujucrashdump/addons.py

```diff
@@ -41,59 +45,90 @@ def temp_function(*args, **kwargs):
         func(*args, **kwargs)
         os.chdir(olddir)
         shutil.rmtree(tempdir)
+
     return temp_function
 
 
-def load_addons(addons_file_path):
+def load_addons(addons_file_path, enabled_addons, as_root):
     with open(addons_file_path) as addons_file:
         addon_specs = yaml.safe_load(addons_file)
     addons = {}
     for name, info in addon_specs.items():
-        try:
-            assert(LOCAL in info)
-            assert(REMOTE in info)
-        except AssertionError:
-            logging.warning(
-                'The addons file: "%s" is malformed, "%s" does not have one'
-                ' of the necessary keys "%s" or "%s"' %
-                (addons_file_path, name, LOCAL, REMOTE))
-            sys.exit(1)
-        addons[name] = CrashdumpAddon(name, info[LOCAL], info[REMOTE])
+        if name not in enabled_addons:
```

Comment:
if a user has to specify the addon, i feel like we can assume they know if its running as root or not

Reply:
If the user has loaded their own addon, yes. The user does however not see the source code of addons from juju crashdump, so they don't know if a addon will use `sudo`.

Also, I moved `run_listening` and `run_psaux` from crashdump to addons.yaml. They are loaded by default for backwards consistency and both use `sudo`.